core/src/main/scala/org/apache/spark/storage/ExternalBlockStore.scala

```diff
@@ -110,7 +110,7 @@ private[spark] class ExternalBlockStore(blockManager: BlockManager, executorId:
         val byteBuffer = bytes.duplicate()
         byteBuffer.rewind()
         externalBlockManager.get.putBytes(blockId, byteBuffer)
-        val size = bytes.limit()
+        val size = bytes.remaining()
```

Comment:
This is an interesting one. The original `bytes` is reset to position 0 and put, so it means that the size that was put really is `bytes.limit` or `byteBuffer.remaining` (before it's put). I think this might have to be adjusted but I'm also not sure why `bytes` is returned here instead of what was put into the block manager. Maybe it's really assumed position is 0, but then why `rewind`?